settings/gitlab.mdx

```diff
@@ -8,30 +8,78 @@ icon: 'gitlab'
   GitLab integration is available on the [Enterprise plan](https://mintlify.com/pricing).
 </Check>
 
-<Steps>
+We use a combination of Access tokens and Webhooks to authenticate and sync changes between GitLab and Mintlify.
+
+* We use Access tokens to pull information from GitLab Mintlify.
+* We use Webhooks so GitLab can notify Mintlify when changes are made.
+  * This allows Mintlify to create preview deployments when a MR is created.
+
+<Steps title="In Mintlify">
+  <Step title="Begin Setup">
+    Within your Mintlify dashboard, click on the `Settings` tab and select [`Git Settings`][git-settings].
+  </Step>
+  <Step title="Switch to GitLab">
+    Click on `Switch to GitLab` which will open a panel to enter your GitLab information.
+  </Step>
+</Steps>
+
+<Steps title="In GitLab">
   <Step title="Find your project ID">
-    Navigate to your project's `Settings` > `General` and find the Project ID.
+    In another tab, navigate to your GitLab's project `Settings` > `General` and find the `Project ID`.
     <Frame>
       <img src="/images/gitlab/gitlab-project-id.png" />
     </Frame>
   </Step>
-  <Step title="Generate a project private token">
-
-    a. Navigate to your project's `Settings` > `Access Tokens`.
+  <Step title="Generate an Access token">
+    a. Navigate to your GitLab's project `Settings` > `Access Tokens`.
 
     b. Select `Add new token`.
-      1. Name the token "Mintlify" - this specific name is important as it will be visible when we make changes on our end.
-      2. Choose `read_api` for the scopes.
-      3. If you have a private repo, you must set the role as `Reporter`.
+      1. Name the token "Mintlify".
+      2. If you have a private repo, you must set the role as `Maintainer`.
+      3. Choose `api` and `read_api` for the scopes.
 
     c. Finally click `Create project access token` and copy the token.
 
     <Frame>
       <img src="/images/gitlab/gitlab-project-access-token.png" />
     </Frame>
+  </Step>
+
+  <Step title="Create the webhook">
+    a. Navigate to your GitLab's project `Settings` > `Webhooks`.
+
+    b. In the "URL" field, enter the endpoint `https://leaves.mintlify.com/gitlab-webhook`
+
+    c. Name the webhook "Mintlify".
 
+    d. Paste the Webhook token generated after completing the previous steps.
```

Comment:
I think we should add a screenshot of the product here highlighting the "Show Webtoken".

Reply:
abs, on it!

Reply:
updated!